File: verify.py
import json

with open("path.json", "r") as f:
    my_dict = json.load(f)
test = my_dict['path']

from PIL import Image

def extract_left_bottom_sector(input_path, output_path):
    # Open the image file
    image = Image.open(input_path)

    # Get the dimensions of the image
    width, height = image.size

    # Calculate coordinates for the left-bottom sector
    left = 0
    top = height // 2  # Top is set to half of the image height
    right = width // 2  # Right is set to half of the image width
    bottom = height  # Bottom is set to the full image height

    # Crop the image using the calculated coordinates
    cropped_image = image.crop((left, top, right, bottom))

    # Save the cropped image to the output path
    cropped_image.save(output_path)

# Example usage:
input_image_path = test;
output_image_path = 'left_bottom_sector.jpg'

# Call the function to extract the left-bottom sector
extract_left_bottom_sector(input_image_path, output_image_path)

File: verify.js
const fs = require('fs');
const sharp = require('sharp');

// Read the JSON file
const jsonString = fs.readFileSync('path.json');
const myDict = JSON.parse(jsonString);
const inputImagePath = myDict.path;

// Function to extract left-bottom sector
async function extractLeftBottomSector(inputPath, outputPath) {
    try {
        // Read the input image
        const inputImageBuffer = await fs.promises.readFile(inputPath);

        // Get the dimensions of the image
        const { width, height } = await sharp(inputImageBuffer).metadata();

        // Calculate coordinates for the left-bottom sector
        const left = 0;
        const top = Math.floor(height / 2); // Top is set to half of the image height
        const right = Math.floor(width / 2); // Right is set to half of the image width
        const bottom = height; // Bottom is set to the full image height

        // Crop and save the image using the calculated coordinates
        await sharp(inputImageBuffer)
            .extract({ left, top, width: right - left, height: bottom - top }) // Fix: Use width and height instead of right and bottom directly
            .toFile(outputPath);
        
        console.log('Image processed successfully.');
    } catch (error) {
        console.error('Error processing image:', error);
    }
}

// Example usage:
const outputImagePath = 'left_bottom_sector.jpg';

// Call the function to extract the left-bottom sector
extractLeftBottomSector(inputImagePath, outputImagePath);
